### PR TITLE
MoMMi-side of an automatic #ick lock after a round ends.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Need also to do a :
- Secret repo side of this (by adding #ick unique identifier to `server.toml` or whatever so that I can call `send2discord` with this identifier in Byond)
- BYOND side of this (send the magic string, which is currently the discord password, to the `#ick` discord channel)

From a cursory look at MoMMi, what the documentation calls a `Guild` is `self.server` within the `MChannel` class, and what the documentation calls a `Channel` is the `discordpy_channel` attribute. 

I used a stackexchange bit of code to find what the `default_role` attribute and `set_permissions` method were.

Please tell me if I did anything wrong.

### CODE REVIEW

**Comment by SonixApache** [comment]
trash idea

**Comment by boy2mantwicethefam** [comment]
Incredibly powerful

**Comment by ShiftyRail** [comment]
@PJB3005 Can you store user-defined variables in `server.toml` or is that file just for giving configuration info
I want to keep the discord-killphrase secret repo side, but I can't find of a way to do it except making it identical to the password

**Comment by jwhitak** [comment]
Discussion on Discord seems to overall favor this but with a shorter duration, 3 minutes has been mentioned a lot. I'm not personally for the PR, but the need for it to be there is understandable. The thread is the lifeblood of the server.

File: MoMMI/Modules/gamenudge.py [hunk 1]
```diff
@@ -23,6 +23,11 @@ async def gamenudge(channel: MChannel, message: Any, meta: str) -> None:
 
     content = content.replace("@", "@\u200B") # Zero-Width space to prevent pings.
 
+    # This string closes the #ick channel for 10 minutes
+    if content == password:
+        thread_reminder(channel)
```
Comment (PJB3005) [inline MoMMI/Modules/gamenudge.py:57]: You're not actually importing this function from the other module.

File: MoMMI/Modules/thread_reminder.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from MoMMI import MChannel
+import time
+
+
+logger = logging.getLogger(__name__)
+
+async def thread_reminder(channel: MChannel) -> None:
+
+    # Send the message
+    await channel.send("**A round has ended.** You can discuss it at https://boards.4chan.org/vg/catalog#s=ss13g. This channel will be closed for 10 minutes.")
```
Comment (PJB3005) [inline MoMMI/Modules/thread_reminder.py]: This says 10 minutes but you lowered it to 5 below.

File: MoMMI/channel.py [hunk 3]
```diff
@@ -127,6 +127,14 @@ def get_role_snowflake(self, snowflake: SnowflakeID) -> Role:
 
         raise ValueError(f"Unknown role {snowflake}")
 
+    # Close the channel
+    def close()
+        await discordpy_channel.set_permissions(self.server.default_role, send_messages=False)
+
+    # Open it
+    def open()
+        await discordpy_channel.set_permissions(self.server.default_role, send_messages=True)
```
Comment (PJB3005) [inline MoMMI/channel.py]:
```suggestion
        await discordpy_channel.set_permissions(self.server.discordpy_server.default_role, send_messages=False)

    # Open it
    def open()
        await discordpy_channel.set_permissions(self.server.discordpy_server.default_role, send_messages=True)
```

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by ShiftyRail** [comment]
Done, also need whatever magic you use to convert the string "ick" into the identifier of the #ick channel.

File: MoMMI/Modules/thread_reminder.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+logger = logging.getLogger(__name__)
+
+async def thread_reminder(channel: MChannel) -> None:
+
+    # Send the message
+    await channel.send("**A round has ended.** You can discuss it at https://boards.4chan.org/vg/catalog#s=ss13g. This channel will be closed for 5 minutes.")
+    # And make it closed
+    await channel.close()
+    # For 5 minutes
+    time.sleep(300)
```
Comment (PJB3005) [inline MoMMI/Modules/thread_reminder.py]: oh I almost forgot, you need to do `await asyncio.delay()` not `time.sleep()` here or else you'll lock up the whole bot.

File: MoMMI/channel.py [hunk 5]
```diff
@@ -127,6 +127,14 @@ def get_role_snowflake(self, snowflake: SnowflakeID) -> Role:
 
         raise ValueError(f"Unknown role {snowflake}")
 
+    # Close the channel
+    def close()
```
Comment (PJB3005) [inline MoMMI/channel.py]:
```suggestion
    async def close():
```

File: MoMMI/channel.py [hunk 6]
```diff
@@ -127,6 +127,14 @@ def get_role_snowflake(self, snowflake: SnowflakeID) -> Role:
...
+        await discordpy_channel.set_permissions(self.server.discordpy_server.default_role, send_messages=False)
+
+    # Open it
+    def open()
```
Comment (PJB3005) [inline MoMMI/channel.py]:
```suggestion
    async def open():
```

**Review by PJB3005 — COMMENTED** [review]
Whoops forgot about these

File: MoMMI/channel.py [hunk 7]
```diff
@@ -127,6 +127,14 @@ def get_role_snowflake(self, snowflake: SnowflakeID) -> Role:
 
         raise ValueError(f"Unknown role {snowflake}")
 
+    # Close the channel
+    async def close()
+        await discordpy_channel.set_permissions(self.server.discordpy_server.default_role, send_messages=False)
+
+    # Open it
+    async def open()
```
Comment (PJB3005) [inline MoMMI/channel.py]: You forgot the colons here.

File: MoMMI/Modules/thread_reminder.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import logging
```
Comment (PJB3005) [inline MoMMI/Modules/thread_reminder.py:1]: And didn't import asyncio here.

**Comment by ShiftyRail** [comment]
Thank you for your patience, PJB

**Pull request merged**